{"content":"---\ntitle: \"Variable '<variablename>' hides a variable in an enclosing block\"\nms.date: 07/20/2015\nf1_keywords: \n  - \"vbc30616\"\n  - \"bc30616\"\nhelpviewer_keywords: \n  - \"BC30616\"\nms.assetid: e7658ebc-da45-451b-a409-a0f8915f0beb\n---\n# Variable '\\<variablename>' hides a variable in an enclosing block\nA variable enclosed in a block has the same name as another local variable.  \n  \n **Error ID:** BC30616  \n  \n## To correct this error  \n  \n-   Rename the variable in the enclosed block so that it is not the same as any other local variables. For example:  \n  \n    ```  \n    Dim a, b, x As Integer  \n    If a = b Then  \n       Dim y As Integer = 20 ' Uniquely named block variable.  \n    End If  \n    ```  \n  \n-   A common cause for this error is the use of `Catch e As Exception` inside an event handler. If this is the case, name the `Catch` block variable `ex` rather than `e`.  \n  \n-   Another common source of this error is an attempt to access a local variable declared within a `Try` block in a separate `Catch` block. To correct this, declare the variable outside the `Try...Catch...Finally` structure.  \n  \n## See also\n\n- [Try...Catch...Finally Statement](../../../visual-basic/language-reference/statements/try-catch-finally-statement.md)\n- [Variable Declaration](../../../visual-basic/programming-guide/language-features/variables/variable-declaration.md)\n","nodes":[{"pos":[4,225],"embed":true,"restype":"x-metadata","content":"title: \"Variable '<variablename>' hides a variable in an enclosing block\"\nms.date: 07/20/2015\nf1_keywords: \n  - \"vbc30616\"\n  - \"bc30616\"\nhelpviewer_keywords: \n  - \"BC30616\"\nms.assetid: e7658ebc-da45-451b-a409-a0f8915f0beb","nodes":[{"content":"Variable '<variablename>' hides a variable in an enclosing block","nodes":[{"pos":[0,64],"content":"Variable '<ph id=\"ph1\">&lt;variablename&gt;</ph>' hides a variable in an enclosing block","source":"Variable '<variablename>' hides a variable in an enclosing block"}],"path":["title"],"nosxs":false}],"yml":true},{"pos":[232,297],"content":"Variable '\\<variablename>' hides a variable in an enclosing block","linkify":"Variable '\\<variablename>' hides a variable in an enclosing block","nodes":[{"content":"Variable '<ph id=\"ph1\">\\&lt;</ph>variablename&gt;' hides a variable in an enclosing block","pos":[0,65],"source":"Variable '\\<variablename>' hides a variable in an enclosing block"}]},{"content":"A variable enclosed in a block has the same name as another local variable.","pos":[298,373]},{"pos":[380,401],"content":"<bpt id=\"p1\">**</bpt>Error ID:<ept id=\"p1\">**</ept> BC30616","source":"**Error ID:** BC30616"},{"pos":[410,431],"content":"To correct this error","linkify":"To correct this error","nodes":[{"content":"To correct this error","pos":[0,21]}]},{"content":"Rename the variable in the enclosed block so that it is not the same as any other local variables.","pos":[441,539]},{"content":"For example:","pos":[540,552]},{"content":"A common cause for this error is the use of <ph id=\"ph1\">`Catch e As Exception`</ph> inside an event handler.","pos":[711,802],"source":"A common cause for this error is the use of `Catch e As Exception` inside an event handler."},{"content":"If this is the case, name the <ph id=\"ph1\">`Catch`</ph> block variable <ph id=\"ph2\">`ex`</ph> rather than <ph id=\"ph3\">`e`</ph>.","pos":[803,877],"source":" If this is the case, name the `Catch` block variable `ex` rather than `e`."},{"content":"Another common source of this error is an attempt to access a local variable declared within a <ph id=\"ph1\">`Try`</ph> block in a separate <ph id=\"ph2\">`Catch`</ph> block.","pos":[887,1022],"source":"Another common source of this error is an attempt to access a local variable declared within a `Try` block in a separate `Catch` block."},{"content":"To correct this, declare the variable outside the <ph id=\"ph1\">`Try...Catch...Finally`</ph> structure.","pos":[1023,1107],"source":" To correct this, declare the variable outside the `Try...Catch...Finally` structure."},{"pos":[1116,1124],"content":"See also","linkify":"See also","nodes":[{"content":"See also","pos":[0,8]}]},{"pos":[1128,1245],"content":"<bpt id=\"p1\">[</bpt>Try...Catch...Finally Statement<ept id=\"p1\">](../../../visual-basic/language-reference/statements/try-catch-finally-statement.md)</ept>","source":"[Try...Catch...Finally Statement](../../../visual-basic/language-reference/statements/try-catch-finally-statement.md)"},{"pos":[1248,1363],"content":"<bpt id=\"p1\">[</bpt>Variable Declaration<ept id=\"p1\">](../../../visual-basic/programming-guide/language-features/variables/variable-declaration.md)</ept>","source":"[Variable Declaration](../../../visual-basic/programming-guide/language-features/variables/variable-declaration.md)"}]}